enterprise/server/backends/distributed/distributed_test.go

```diff
@@ -285,10 +285,10 @@ func TestEventualConsistency(t *testing.T) {
 		restartCache := func(peer string) {
 			log.Printf("Restarting %q...", peer)
 			mu.Lock()
-			defer mu.Unlock()
 			peers = append(peers, peer)
 			caches[peer] = newDistributedCache(t, te, peer, testStruct.replicationFactor, maxSizeBytes)
 			waitForNodes(peers)
+			mu.Unlock()
```

Comment:
we cannot do the same thing we did above here because the peers list is our reference for nodes to send requests to. If we restart a node and add it to peers, it may not yet know about any of the rest of the system, so sending requests to it is futile.